dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/put-fee-record-correction-transient-form-data.controller/index.test.ts

```diff
@@ -0,0 +1,199 @@
+import httpMocks from 'node-mocks-http';
+import { ObjectId } from 'mongodb';
+import {
+  FeeRecordCorrectionTransientFormDataEntity,
+  RECORD_CORRECTION_REASON,
+  RecordCorrectionTransientFormData,
+  REQUEST_PLATFORM_TYPE,
+  TestApiError,
+} from '@ukef/dtfs2-common';
+import { HttpStatusCode } from 'axios';
+import { aTfmSessionUser } from '../../../../../../test-helpers';
+import { PutFeeRecordCorrectionTransientFormDataRequest, putFeeRecordCorrectionTransientFormData } from '.';
+import { PutFeeRecordCorrectionTransientFormDataPayload } from '../../../../routes/middleware/payload-validation';
+import { FeeRecordCorrectionTransientFormDataRepo } from '../../../../../repositories/fee-record-correction-transient-form-data-repo';
+import { FeeRecordRepo } from '../../../../../repositories/fee-record-repo';
+
+jest.mock('../../../../../repositories/fee-record-repo');
+jest.mock('../../../../../repositories/fee-record-correction-transient-form-data-repo');
+
+console.error = jest.fn();
+
+describe('put-fee-record-correction-transient-form-data.controller', () => {
+  describe('putFeeRecordCorrectionTransientFormData', () => {
+    const reportId = 1;
+    const feeRecordId = 2;
+
+    const tfmUserId = new ObjectId().toString();
+
+    const formData: RecordCorrectionTransientFormData = {
+      reasons: [RECORD_CORRECTION_REASON.FACILITY_ID_INCORRECT, RECORD_CORRECTION_REASON.OTHER],
+      additionalInfo: 'Some additional info',
+    };
+
+    const aValidRequestQuery = () => ({ reportId: reportId.toString(), feeRecordId: feeRecordId.toString() });
+
+    const aValidRequestBody = (): PutFeeRecordCorrectionTransientFormDataPayload => ({
+      user: { ...aTfmSessionUser(), _id: tfmUserId },
+      formData,
+    });
+
+    const mockFeeRecordExists = jest.fn();
+    const mockSave = jest.fn();
+
+    beforeEach(() => {
+      FeeRecordRepo.existsByIdAndReportId = mockFeeRecordExists;
+      mockFeeRecordExists.mockReturnValue(true);
+
+      FeeRecordCorrectionTransientFormDataRepo.save = mockSave;
+      mockSave.mockResolvedValue(undefined);
+    });
+
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it(`responds with a '${HttpStatusCode.Ok}' if the transient form data is saved successfully`, async () => {
+      // Arrange
+      const req = httpMocks.createRequest<PutFeeRecordCorrectionTransientFormDataRequest>({
```

Comment:
req and res could go in a beforeEach with a let req and let res at the top as repeated many times

Reply:
Good point, moved these thanks